src/Crypto/RNG.hs

```diff
@@ -81,7 +83,7 @@ randomBytesIO :: ByteLength -- ^ number of bytes to generate
               -> IO ByteString
 randomBytesIO n (CryptoRNGState gv) = do
   liftIO $ modifyMVar gv $ \g -> do
-    (bs, g') <- either (fail "Crypto.GlobalRandom.genBytes") return $
+    (bs, g') <- either (const (fail "Crypto.GlobalRandom.genBytes")) return $
```

Comment:
Why is `const` needed here? :thinking:

Reply:
There is no `MonadFail` instance for `->`.

Reply:
Simply calling `error` will be better then :thinking: That's what fail for IO does anyway.

Reply:
Actually, `fail` in case of IO calls `failIO` which then uses `raiseIO#` as opposed to `error` which uses `raise#`, so I'm not really sure if they are equivalent.

Reply:
You're right in that in general they aren't equivalent because:
- `failIO "wut"` has type `IO a` and is a value that will throw when **executed in the IO monad**.
- `error "wut"` has type `a` and is a value that will throw when **evaluated**.

Which basically means this:

```haskell
λ> failIO "wut" `seq` ()
()
λ> error "wut" `seq` ()
*** Exception: wut
CallStack (from HasCallStack):
  error, called at <interactive>:9:1 in interactive:Ghci2
```

but, when used as a part of monadic block, `error` can be used instead and both will have the same result, i.e. throw. Actually `error` is strictly better here because is has `HasCallStack` constraint, whereas `failIO` doesn't.

Reply:
I had another look at the code and I think we can replace this whole thing with `genBytes` function from `Crypto-Classes-Exceptions` which throws error instead of returning `Either`.

I'm not sure if we should do it in this PR, though.